src/window_identifier/mod.rs

```diff
@@ -215,20 +215,25 @@ impl WindowIdentifier {
     #[cfg(feature = "wayland")]
     /// Create an instance of [`WindowIdentifier`] from a Wayland surface.
     ///
-    /// ## Safety
+    /// # Safety
     ///
-    /// The surface has to be a valid Wayland surface pointer.
-    pub unsafe fn from_wayland_raw(surface_ptr: *mut std::ffi::c_void) -> Self {
-        match WaylandWindowIdentifier::from_raw(surface_ptr) {
+    /// Both pointers have to be valid surface and display pointers. You must
+    /// ensure the `display_ptr` lives longer than the returned
+    /// `WindowIdentifier`.
+    pub async unsafe fn from_wayland_raw(
```

Comment:
I'm actually considering whether this function is necessary. Supporting this function introduces a bunch of safety requirements and lifetime questions we need to ask about the connection.

I think what we can do is explain where the user would create the required types from the surface and display pointers (wayland-backend) and then explain what lifetime constraints apply to such handles `WaylandWindowIdentifier` has (must not outlive the connection).

This would also mean the `client_system` feature would not be needed as ashpd would not be responsible for that and would operate on wayland-backend common types or wayland-client types.

Reply:
Actually no we have to have this, RawWindowHandle forces our hand here

Reply:
> Actually no we have to have this, RawWindowHandle forces our hand here

Yes :(